src/celeritas/ext/detail/CelerEmStandardPhysics.cc

```diff
@@ -435,6 +461,67 @@ void CelerEmStandardPhysics::add_e_processes(G4ParticleDefinition* p)
     }
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Add EM processes for muons.
+ *
+ * | Processes                    | Model classes                |
+ * | ---------------------------- | ---------------------------- |
+ * | Pair production              | G4MuPairProductionModel      |
+ * | Ionization (low E, mu-)      | G4ICRU73QOModel              |
+ * | Ionization (low E, mu+)      | G4BraggModel                 |
+ * | Ionization (high E)          | G4MuBetheBlochModel          |
+ * | Bremsstrahlung               | G4MuBremsstrahlungModel      |
+ * | Coulomb scattering           | G4eCoulombScatteringModel    |
+ * | Multiple scattering          | G4WentzelVIModel             |
+ *
+ * \note Currently all muon processes are disabled by default
+ */
+void CelerEmStandardPhysics::add_mu_processes(G4ParticleDefinition* p)
+{
+    auto* physics_list = G4PhysicsListHelper::GetPhysicsListHelper();
+
+    if (options_.mu_pair_production)
+    {
+        physics_list->RegisterProcess(new G4MuPairProduction(), p);
+        CELER_LOG(debug) << "Loaded muon pair production with "
+                            "G4MuPairProductionModel";
+    }
+
+    if (options_.mu_ionization)
+    {
+        physics_list->RegisterProcess(new G4MuIonisation(), p);
+        CELER_LOG(debug) << "Loaded muon ionization with G4ICRU73QOModel, "
+                            "G4BraggModel, and G4MuBetheBlochModel";
+    }
+
+    if (options_.mu_bremsstrahlung)
+    {
+        physics_list->RegisterProcess(new G4MuBremsstrahlung(), p);
```

Comment:
I don't know if this is the right time to address this, but it is worth mentioning: The applicable energy ranges for muon bremsstrahlung are a bit tricky in G4.
- The `G4EmStandardPhysics` only enables muon brems if the `G4EmParameters::MaxKinEnergy() > G4HadronicParamaters::EnergyThresholdForHeavyHadrons()` (which is 1.1 GeV by default)
- The `G4MuBremsstrahlungModel` also has its own applicability that we may double check.

So should we register the process no matter what and its use is defined downstream, or should we make `CelerEmStandardPhysics` more similar to `G4EmStandardPhysics` (i.e. adding thresholds from the get go)?

Reply:
Interesting... I don't think we account for that `MaxKinEnergy` being anything other than the default in our code currently (e.g. we just hardcode all our model upper limits to the dafault 100 TeV) so maybe it's fine not worrying about it for now, but I'm also happy to add it. Out of curiosity, do you know in what circumstances `MaxKinEnergy` mght be set to something lower? Looks like `EnergyThresholdForHeavyHadrons` has a maximum possible value of 5 GeV, and its description says:
> if max kinetic energy is below this limit EM and hadronic physics is not instantiated for hyperons, anti-hyperons, anti light ions, b-, c- particles

For the model applicability, since it's an energy loss process we should be ok letting the limits be defined implicitly by the bounds of the energy loss/cross section tables (which will be determined by the prodution cuts and `G4MuBremsstrahlungModel::MinEnergyCut()`).

Reply:
Yep. I also assumed it being ok since the XS tables would limit it, and I tend to prefer registering here and applying limits downstream. At least to me, this seems like a more logical approach: the physics list sets the processes on or off. If there are constraints, these are applied later. Regarding your question about the `MaxKinEnergy()`, that I don't know for sure. I would guess that some detectors saturate above a certain energy, so you just make the simulation faster by reducing that for a set of models? Maybe @whokion knows...

Reply:
Well, the brem cross section of muons is `relatively` small compared to ionization at the low energy and the radiative effect becomes meaningful (%level) as a part of the total energy loss only at the high energy (order of GeV or higher, but depending on materials).